packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -196,10 +202,56 @@ export default class CallDiagnosticMetrics {
   }
 
   /**
-   * TODO: NOT IMPLEMENTED
+   * Return Client Event payload by client error code
+   * @param clientErrorCode
+   * @returns
+   */
+  public getErrorPayloadForClientErrorCode(clientErrorCode: number): ClientEventError {
+    let error: ClientEventError;
+
+    if (clientErrorCode) {
+      const partialParsedError = CLIENT_ERROR_CODE_TO_ERROR_PAYLOAD[clientErrorCode];
+
+      if (partialParsedError) {
+        error = merge(
+          {fatal: true, shownToUser: false, name: 'other', category: 'other'}, // default values
+          partialParsedError
+        );
+
+        return error;
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Generate error payload for Client Event
    * @param rawError
    */
-  generateErrorPayload(error: any) {
+  generateClientEventErrorPayload({
+    rawError,
+    parsedError,
+  }: {
+    rawError?: any;
+    parsedError?: ClientEventError;
+  }) {
+    // sometimes we compute the error on the go
+    if (parsedError) {
+      return parsedError;
+    }
+
+    const errorCode = rawError?.body?.errorCode || rawError?.body?.code;
+    if (errorCode) {
+      const clientErrorCode = SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP[errorCode];
+      if (clientErrorCode) {
+        return this.getErrorPayloadForClientErrorCode(clientErrorCode);
+      }
+
+      // by default, send new locus error
+      return this.getErrorPayloadForClientErrorCode(NEW_LOCUS_ERROR_CLIENT_CODE);
+    }
+
     return undefined;
```

Comment:
from debug point of view it would be nice to have a fallback "Unknown" error.

Reply:
When we can't figure out the error, it's better to return undefined. This will make sure that `errors` property on the event payload doesn't get set. This is important for CA team, it's safer for us to send nothing if the structure of the error is incomplete, as it will fail on their side.

`clearEmptyKeysRecursively` takes care of this after preparing the event.

Reply:
More information (especially about failure) better then less, but if it is a requirement from the CA team then fine